packages/react-router/src/router.ts

```diff
@@ -3145,9 +3166,9 @@ ${jsesc(script, { quotes: 'backtick' })}\`)`
       'Key has already been streamed: ' + key,
     )
 
-    this.streamedKeys.add(key)
+    this.emit({ type: 'onStreamedValue', key })
     this.injectScript(
-      `__TSR__.streamedValues['${key}'] = { value: ${this.serializer?.(this.options.transformer.stringify(value))}}`,
+      `__TSR__.streamedValues.push([${this.serializer?.(key)}, ${this.serializer?.(this.options.transformer.stringify(value))}])`,
```

Comment:
`key` was not serialized before, so keys containing `'` could actually cause JavaScript errors or code injection.